packages/amplify_api/test/amplify_api_graphql_helpers_test.dart

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'resources/Blog.dart';
+import 'resources/ModelProvider.dart';
+
+void main() {
+  test("ModelQueries.get() should craft a valid request", () {
+    AmplifyAPI api = AmplifyAPI(modelProvider: ModelProvider.instance);
+
+    String id = UUID.getUUID();
+    String expected =
+        "query getBlog(\$id: ID!) { getBlog(id: \$id) { id name createdAt } }";
+
+    GraphQLRequest<Blog> req = ModelQueries.get<Blog>(Blog.classType, id);
+
+    expect(req.document, expected);
+    expect(req.variables.containsValue(id), true);
```

Comment:
Let's be more specific so we can make sure that `variables` contains everything that's necessary and nothing that's not. If map comparison is failing, try the `equals` matcher or use the `mapEquals` function from Flutter but this can only be used in integration tests. There is also `MapEquality` from package [collection](https://pub.dev/packages/collection) which can be used for very customizable Map comparison and can be used in unit tests.

```dart
expect(req.variables, equals({'id': id}));

// or
import 'package:flutter/foundation.dart';

expect(mapEquals(req.variables, {'id': id}), isTrue);

// or
import 'package:collection/collection.dart';

expect(const MapEquality().equals(req.variables, {'id': id}), isTrue);
```

Reply:
Thanks for the tip! I suspected there was a better way to write assertions for these.